apis/Google.Cloud.Storage.V1/Google.Cloud.Storage.V1/UrlSigner.cs

```diff
@@ -255,6 +282,34 @@ public async Task<string> SignAsync(string bucket, string objectName, TimeSpan d
             GetEffectiveSigner(GaxPreconditions.CheckNotNull(options, nameof(options)).SigningVersion).SignAsync(
                 GaxPreconditions.CheckNotNull(requestTemplate, nameof(requestTemplate)), options, _blobSigner, _clock, cancellationToken);
 
+        /// <summary>
+        /// Signs the given post policy. The result can be used to make form posting requests matching the conditions
+        /// set in the post policy.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// Signing post policies is not supported by <see cref="SigningVersion.V2"/>. A <see cref="NotSupportedException"/>
+        /// will be thrown if an attempt is made to sign a post policy using <see cref="SigningVersion.V2"/>.
+        /// </para>
+        /// <para>
+        /// When a <see cref="UrlSigner"/> is created with a service account credential, the signing can be performed
+        /// with no network access. When it is created with an implementation of <see cref="IBlobSigner"/>, that may require
+        /// network access or other IO. In that case, one of the asynchronous methods should be used when the caller is
+        /// in a context that should not block.
+        /// </para>
+        /// <para>
+        /// See https://cloud.google.com/storage/docs/xml-api/post-object for more information on signed post policies.
+        /// </para>
+        /// </remarks>
+        /// <param name="postPolicy">The post policy to signed and that will be enforced when making the post request.
+        /// Most not be null.</param>
+        /// <param name="options">The options used to generate the signed post policy. Must not be null.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>The signed post policy, which contains all the fields that should be including in the form to post.</returns>
+        public Task<SignedPostPolicy> SignAsync(PostPolicy postPolicy, Options options, CancellationToken cancellationToken = default) =>
+            GetEffectiveSigner(GaxPreconditions.CheckNotNull(options, nameof(options)).SigningVersion).SignAsync(
+                GaxPreconditions.CheckNotNull(postPolicy, nameof(postPolicy)), options, _blobSigner, _clock, cancellationToken);
+
```

Comment:
@amanda-tarafa could you add examples how the client is used in Tests OR [This document](https://docs.google.com/document/d/1jck8zmQVBVTMY09QCfHKdh7XiY0-qzeDC6uKo1F3J6g/edit?hl=en#heading=h.8t1aqb78dzr)?

Reply:
Snippets are coming!